src/playground/components/CodeEditor.js

```diff
@@ -10,33 +10,72 @@ import {
 } from "../utils/codemirror-theme";
 import "../scss/editor.scss";
 
-export default function CodeEditor({
-	codeValue,
-	onUpdate,
-	eslintOptions,
-	eslintInstance,
-}) {
-	const extensions = useMemo(
-		() => [
-			history(),
-			bracketMatching(),
-			linter(esLint(eslintInstance, eslintOptions), { delay: 0 }),
-			javascript(),
-			ESLintPlaygroundTheme,
-			ESLintPlaygroundHighlightStyle,
-		],
-		[eslintOptions, eslintInstance],
-	);
-
-	return (
-		<CodeMirror
-			value={codeValue}
-			minWidth="100%"
-			height="100%"
-			extensions={extensions}
-			onChange={value => {
-				onUpdate(value);
-			}}
-		/>
-	);
-}
+const CodeEditor = forwardRef(
+	({ codeValue, onUpdate, eslintOptions, eslintInstance }, ref) => {
+		const editorRef = useRef(null);
+
+		const extensions = useMemo(
+			() => [
+				history(),
+				bracketMatching(),
+				linter(esLint(eslintInstance, eslintOptions), { delay: 0 }),
+				javascript(),
+				ESLintPlaygroundTheme,
+				ESLintPlaygroundHighlightStyle,
+			],
+			[eslintOptions, eslintInstance],
+		);
+
+		useImperativeHandle(ref, () => ({
+			scrollToPosition(line, col) {
+				if (editorRef.current) {
+					const editorView = editorRef.current.view;
+					if (editorView) {
+						const { state } = editorView;
+						const pos = state.doc.line(line).from + col;
```

Comment:
Column where the cursor is after clicking the 2:11 link seems to be off-by-one:

![image](https://github.com/eslint/eslint.org/assets/44349756/046bc0c2-0c98-4c9e-875e-a31c4c49d68e)

Note that in ESLint messages both lines and columns are 1-based.

Reply:
so in this case it should be after the b character?

Reply:
The other way, it should be before the left quote.